staging/src/k8s.io/client-go/tools/leaderelection/leaderelection.go

```diff
@@ -187,6 +188,9 @@ type LeaderElector struct {
 	// clock is wrapper around time to allow for less flaky testing
 	clock clock.Clock
 
+	// used to lock the observedRecord
+	mutex sync.RWMutex
```

Comment:
please make the name more specific, e.g. `observedRecordLock`

Reply:
Thanks! I fixed: https://github.com/kubernetes/kubernetes/pull/97419/commits/301a7bdd497b93a3fa45cf2f9a2173cd7a2ed376